### PR TITLE
Fake: copy over grad attribute

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82593
* #82571
* #82337
* #82449

Copy over the grad attribute... let me any suggestions of other tests/opinfos/crossrefs I should be doing to make this more comprehensive.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/_subclasses/fake_tensor.py [hunk 1]
```diff
@@ -137,6 +137,8 @@ def from_real_tensor(self, fake_mode, t):
             out = FakeTensor(fake_mode, self.meta_converter(t), existing_device)
         if type(t) is torch.nn.Parameter:
             out = torch.nn.Parameter(out, requires_grad=out.requires_grad)  # type: ignore[assignment]
+        if safe_is_leaf(t) and t.grad is not None:
+            out.grad = self.from_real_tensor(fake_mode, t.grad)
```
Comment (ezyang) [inline torch/_subclasses/fake_tensor.py:141]: what if we have grad on a non-leaf tensor 🤔

Comment (eellison) [reply torch/_subclasses/fake_tensor.py:141]: TBH, i'm not really sure...

Comment (ezyang) [reply torch/_subclasses/fake_tensor.py:141]: I think I'd just drop the leaf check

```
>>> torch.zeros(2, requires_grad=True)
tensor([0., 0.], requires_grad=True)
>>> x = torch.zeros(2, requires_grad=True)
>>> y = x * 2
>>> y.grad = torch.zeros(2)
>>> y
tensor([0., 0.], grad_fn=<MulBackward0>)
>>> y.grad
tensor([0., 0.])
```

**Pull request closed without merging**